backend/src/filter/validateMatchAnyCondition.ts

```diff
@@ -0,0 +1,23 @@
+import { ArgumentError } from '../shared/errors';
+import { FilterCondition, MatchAnyCondition } from '../shared/queries';
+import { formatErrorMessage } from './formatErrorMessage';
+import { checkValidMatchAllAnyCondition } from './checkValidMatchAllAnyCondition';
+
+export function validateMatchAnyCondition(
+  condition: FilterCondition
+): MatchAnyCondition {
+  const result = <MatchAnyCondition>condition;
```

Comment:
Why is there an instant type change? So why is ne argument type not of type `MatchAnyCondition`?

Reply:
Thats because the function does in fact a protected cast. The check is in the next line.